xla/service/gpu/cudnn_fused_mha_transpose_fusion.cc

```diff
@@ -137,9 +140,7 @@ StatusOr<bool> FuseArgPrologueTransposeWithcuDNNFMHA(
   absl::Span<const int64_t> minor_to_major_bmm =
       transpose_arg_operand->shape().layout().minor_to_major();
   if ((minor_to_major_bmm[0] != new_bmm_checked_dims[0]) &&
-      ((transpose_arg_operand->shape().dimensions().at(
-            new_bmm_checked_dims[0]) == 64) ||
-       (IsBwdCustomCallTofMHA(*fmha) && operand_index == 3))) {
+      !(IsBwdCustomCallTofMHA(*fmha) && operand_index == 3)) {
```

Comment:
for regular non flash attention, the head dim has to be 64 and be the fastest dim. for flash attention, the head dim does not have to be 64(it can be also be 128). Also check fastest dim is 64 or not is not good enough, sequence length can also be 64. The right check would be use the contracting dim of bmm1 as the head dim. I think I remove this constraint as WAR to trigger flash attention fusion but didn't add back the more detailed check here.

Reply:
So after taking the second look, the current checks should work as expected. Since all arguments to fmha calls are already transposed to a layout such that head dim is the fastest dim in the rewriter pass. The checks here only need to make sure head dim is still the fastest before the transpose(this transpose is not the transpose introduced in rewriter, could be other transpose introduced by the framework). `minor_to_major_bmm[0]` is the fastest dim and `new_bmm_checked_dims[0]` is the head dim before transpose. It does not need to be 64, so removing that is necessary.